ingest/ledgerbackend/ledger_store.go

```diff
@@ -0,0 +1,54 @@
+package ledgerbackend
+
+import (
+	sq "github.com/Masterminds/squirrel"
+	"github.com/stellar/go/support/db"
+)
+
+// Ledger contains information about a ledger (sequence number and hash)
+type Ledger struct {
+	Sequence uint32 `db:"sequence"`
+	Hash     string `db:"ledger_hash"`
+}
+
+// LedgerStore is used to query ledger data from the Horizon DB
+type LedgerStore interface {
```

Comment:
Nit: emphasise that trust is required.
```suggestion
// TrustedLedgerStore is used to query ledger data from a trusted source.
// The store should contain ledgers verified by Stellar-Core, do not use untrusted
// source like history archives.
type TrustedLedgerStore interface {
```